src/utils/validation/getFieldRules.js

```diff
@@ -1,31 +1,33 @@
-import path from 'ramda/src/path'
-
-/**
- * Returns the collection of validation rules of the given selector
- * applicable to the given field.
- */
-export const getRulesBySelector = (selector, fieldProps, applicableRules) => {
-  const keyPath = [selector, fieldProps[selector]]
-
-  /**
-   * @todo Shallow keyed collection is not a usual behavior, but only suitable
-   * for the reduced schema into "applicableRules". Consider a unified interface.
-   */
-  return path(keyPath, applicableRules)
-}
+import * as R from 'ramda'
+import { getRulesPaths } from '../formUtils/filterSchemaByField'
 
 /**
  * Returns "type" and "name" groups of validation rules
  * relevant to the given field.
+ *
+ * @returns
+ * {
+ *  fieldPath: {
+ *    type: [ValidationRule],
+ *    name: [ValidationRule],
+ *  },
+ *  type: [ValidationRule],
+ *  name: [ValidationRule],
+ * }
  */
 export default function getFieldRules(fieldProps, applicableRules) {
-  return ['type', 'name'].reduce((rules, selector) => {
-    const rulesGroup = getRulesBySelector(selector, fieldProps, applicableRules)
+  const rulesPaths = getRulesPaths(fieldProps)
+
+  return rulesPaths.reduce((acc, ruleKeyPath) => {
+    const selector = R.head(ruleKeyPath)
 
-    if (rulesGroup) {
-      rules[selector] = rulesGroup
-    }
+    /** @todo This is weird */
+    const keyPath =
+      selector === 'fieldPath'
+        ? [selector, ruleKeyPath[ruleKeyPath.length - 2]]
+        : [selector]
 
-    return rules
+    const ruleGroups = R.path(ruleKeyPath, applicableRules)
+    return ruleGroups ? R.assocPath(keyPath, ruleGroups, acc) : acc
```

Comment:
The necessity to stitch the rules in `filterSchemaByField` and then lookup relevant rules here seems so unreasonable. I wonder, can it just reference the key paths from `getRulesPaths` in the `applicableRules` directly?

There is an overhead in composing/decomposing these kind of structures.

Reply:
Will be refined later due to time constrains.